File: backend/routers/notes.py
"""笔记相关路由"""
from fastapi import APIRouter, HTTPException, Query
from services.file_service import get_note_tree, get_note_content, search_notes

router = APIRouter(prefix="/api/notes", tags=["notes"])


@router.get("")
async def list_notes():
    """获取笔记目录树"""
    tree = get_note_tree()
    return {
        "success": True,
        "data": tree
    }


@router.get("/search")
async def search(q: str = Query(..., min_length=1, description="搜索关键词")):
    """搜索笔记"""
    results = search_notes(q)
    return {
        "success": True,
        "data": results,
        "total": len(results)
    }


@router.get("/{note_path:path}")
async def read_note(note_path: str):
    """获取指定笔记的内容"""
    content = get_note_content(note_path)
    
    if content is None:
        raise HTTPException(status_code=404, detail="笔记不存在")
    
    return {
        "success": True,
        "data": {
            "path": note_path,
            "content": content
        }
    }

File: backend-go/routes/notes.go
package routes

import (
	"net/http"

	"mdxy-backend/services"

	"github.com/gin-gonic/gin"
	"gorm.io/gorm"
)

// registerNotesRoutes 注册笔记相关路由
func registerNotesRoutes(router *gin.Engine, db *gorm.DB) {
	notesGroup := router.Group("/api/notes")

	// 获取笔记目录树
	notesGroup.GET("", func(c *gin.Context) {
		tree := services.GetNoteTree()
		c.JSON(http.StatusOK, gin.H{
			"success": true,
			"data":    tree,
		})
	})

	// 搜索笔记
	notesGroup.GET("/search", func(c *gin.Context) {
		q := c.Query("q")
		if q == "" {
			c.JSON(http.StatusBadRequest, gin.H{
				"success": false,
				"message": "搜索关键词不能为空",
			})
			return
		}

		results := services.SearchNotes(q)
		c.JSON(http.StatusOK, gin.H{
			"success": true,
			"data":    results,
			"total":   len(results),
		})
	})

	// 获取指定笔记的内容 - 使用参数化路由而不是通配符
	notesGroup.GET("/:notePath", func(c *gin.Context) {
		notePath := c.Param("notePath")

		// 处理嵌套路径
		if c.Query("path") != "" {
			notePath = c.Query("path")
		}

		content := services.GetNoteContent(notePath)
		if content == "" {
			c.JSON(http.StatusNotFound, gin.H{
				"success": false,
				"message": "笔记不存在",
			})
			return
		}

		c.JSON(http.StatusOK, gin.H{
			"success": true,
			"data": gin.H{
				"path":    notePath,
				"content": content,
			},
		})
	})
}
